sklearn/utils/_response.py

```diff
@@ -94,10 +95,14 @@ def _get_response_values(
                         "classifier with two classes."
                     )
                     raise ValueError(err_msg)
+            elif target_type == "multilabel-indicator" and isinstance(y_pred, list):
```

Comment:
This is worth a comment. From what I found in https://github.com/scikit-learn/scikit-learn/issues/19880, you already have the compressed format that this transform is doing. This would be True for MLP and Ridge classifiers.

Reply:
Let's say `y_pred` is not a list and an `ndarray`, should we throw an error stating that this is not expected for mutilabel targets?